File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.java
class Solution {
    public int largestRectangleArea(int[] heights) {
       int n=heights.length;
       int max=0;
       for(int i=0;i<n;i++){
        int h=heights[i];
        int right=i+1;
        while(right<n && heights[right]>=h){
            right++;
        }
        int left=i;
        while(left>=0 && heights[left]>=h){
            left--;
        }
        right--;
        left++;
        max=Math.max(max,h*(right-left+1));
       } 
       return max;
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        d=0
        def height( root: Optional[TreeNode]) -> int:
            nonlocal d
            if not root:
                return 0
            l=height(root.left)
            r=height(root.right)
            d=max(d,l+r)
            return max(l,r)+1
        height(root)
        return d


        